test/unit/domains.server.test.ts

```diff
@@ -0,0 +1,221 @@
+import { createHostedZone } from '~/lib/dns.server';
+import {
+  createUserDomain,
+  setMonthsFromNow,
+  deleteUserDomain,
+  willExpireIn,
+  isExpired,
+  removeIfExpired,
+  updateUserDomain,
+} from '~/lib/domains.server';
+import { prisma } from '~/db.server';
+import { RecordType } from '@prisma/client';
+
+import type { Domain } from '../../app/lib/domains.server';
+
+describe('Client module function test', () => {
+  const rootDomain = 'starchart.com';
+  const username = 'starchartdev';
+  const description = 'This is to test';
+  const course = 'OSD700';
+  const ports = '8080';
+  const randomWord = (function () {
+    let seed = Date.now();
+    return () => String(seed++);
+  })();
+
+  beforeAll(async () => {
+    process.env.AWS_ROUTE53_HOSTED_ZONE_ID = await createHostedZone(rootDomain);
+  });
+
+  afterAll(() => resetHostedZone());
+
+  test('createUserDomain() creates a record in Route 53 and DB', async () => {
+    const subDomain = randomWord();
+    const domain = `${subDomain}.${rootDomain}`;
+    const value = '192.168.0.1';
+    const data: Domain = {
+      type: RecordType.A,
+      name: domain,
+      value,
+      username,
+    };
+
+    const result = await createUserDomain(data);
+
+    expect(result.name).toEqual(domain);
+    expect(result.type).toEqual('A');
+    expect(result.value).toEqual(value);
+    expect(result.createdAt).not.toBeNaN();
+    expect(result.updatedAt.getTime()).toEqual(result.createdAt.getTime());
+    expect(result.expiresAt.getTime()).toBeGreaterThan(result.createdAt.getTime());
+    expect(result.status.length).toBeGreaterThan(0);
+  });
+
+  test('createUserDomain() throws when record with name, type, value already exists', async () => {
```

Comment:
If your test relies on pre-existing state, that's a red flag.  You should create the state you expect inside this test (i.e., create the record first, then try to create it again).